### DISCUSSION TITLE
Changes in the Matrix side of chatting

### DISCUSSION THREAD

**Discussion by yajo** [OCA/odoo-community.org#102] [Show and tell]
Hello dear community.

Together @francesco-ooops and me have been lately revisiting the state of the chat communities.

## Current situation

As you might know, currently we have [a Discord server](https://discord.gg/rN5xRdE) which is bridged to [Matrix](https://matrix.org/) using [t2bot](https://t2bot.io/discord/), so people using Matrix can join the conversation in [OCA's Matrix space](https://matrix.to/#/#oca:matrix.org).

We are an open FLOSS community, so anybody can join those places to chat. This, of course, is very welcoming for spammers, and we have seen lots of spam. By having the bridge in place, it's even more annoying that Matrix spam reaches Discord users, and viceversa.

## What is changing

@simahawk has been tweaking the Discord side to reduce spam there. The Matrix side is a bit more complex because it doesn't have a built-in spam filter. What people usually do is to [deploy a moderation bot called Mjonlir](https://matrix.org/docs/communities/moderation/) and subscribe it to some ban lists.

To avoid having yet one more deployment to maintain, we have decide to follow a simpler route. Please note that all changes are only for the Matrix side:

1. The OCA space will now be invite-only, instead of public.
2. All rooms within that space (which are bridged to Discord rooms) will require new users to belong to the space before joining the room.
3. There will be [one single new public room for newcomers](https://matrix.to/#/#oca-knock:matrix.org). You can come, introduce yourself and ask for access to the community. One moderator will invite you to the space and then kick you out from that welcome room.
4. Once you're invited into the space, you're able to browse and join any rooms within.

We hope that with this little change, spammers are only able to reach the welcome room, where there will be nobody subscribed except moderators. Then, we can ban spammers there so they don't bother the rest of us. If some of them can still pass that initial barrier and starts spamming, we can also ban them globally in the space.

Spam will keep existing, but we hope this simple change reduces it without impacting too much in how things are currently.

## Future improvements

At the moment, synapse (the server that runs matrix.org) already has the knock feature implemented: https://github.com/matrix-org/synapse/pull/6739. However, the main client (Element) still lacks support for it: https://github.com/vector-im/element-web/issues/18655. Once that is fixed, hopefully we can configure that feature in all rooms and the space can again be open to the public. That'll be even better because each room will be able to have its own moderators. It makes sense that I only moderate the Spanish localization room because I don't participate in any other localization, but others moderate the others'.

Also quoting from [Matrix docs](https://matrix.org/docs/communities/moderation/#getting-mjolnir):

> At the moment, to get mjolnir for your community you need either to pay someone to run mjolnir for you, or to [run it yourself on your own infrastructure](https://github.com/matrix-org/mjolnir/tree/main/docs) if you have the technical knowledge to do it. We're aware this situation is suboptimal and are working on a service so people get can mjolnir bots much more easily.

So hopefully some day there'll be an easier way to introduce a Mjonlir bot in the rooms without having to host it.

## How can you help

Currently we only have 2 moderators in the Matrix space: @francesco-ooops and me. Join us!

If you maintain a Mjonlir bot (or are willing to) and want to give us permissions to use it, please raise your hand.

**Comment by legalsylvain** [comment]
:100: !!!
I left the matrix chat 1 week after registering, because of the spam.
So a huge thank you for working on this!

**Comment by yajo** [comment]
I just wanted to announce that the work is done. Special thanks to @francesco-ooops who did most of it!

**Comment by simahawk** [comment]
@yajo @francesco-ooops super cool, tnx for your help!

How's going here? Can we close this together with https://github.com/orgs/OCA/discussions/104 ?

**Reply by francesco-ooops** [reply]
@simahawk spam is now reduced to 0, so yes I guess you can close them both
